shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -431,10 +440,33 @@ protected FlutterEngine createFlutterEngine(@NonNull Context context) {
     return null;
   }
 
+  /**
+   * Configures a {@link FlutterEngine} after its creation.
+   * <p>
+   * This method is called after the given {@link FlutterEngine} has been attached to the
+   * owning {@code FragmentActivity}. See
+   * {@link io.flutter.embedding.engine.plugins.activity.ActivityControlSurface#attachToActivity(Activity, Lifecycle, PlatformViewsController)}.
+   * <p>
+   * It is possible that the owning {@code FragmentActivity} opted not to connect itself as
+   * an {@link io.flutter.embedding.engine.plugins.activity.ActivityControlSurface}. In that
+   * case, any configuration, e.g., plugins, must not expect or depend upon an available
+   * {@code Activity} at the time that this method is invoked.
+   * <p>
+   * The default behavior of this method is to defer to the owning {@code FragmentActivity}
+   * as a {@link FlutterEngineConfigurator}. Subclasses can override this method if the
+   * subclass needs to override the {@code FragmentActivity}'s behavior, or add to it.
+   */
+  protected void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
+    FragmentActivity attachedActivity = getActivity();
```

Comment:
So this is an auxiliary API to give more control to the activity in case the fragment wants to give the activity to a platfromviewcontroller and the activity says no?

Reply:
Talked offline, resolving old comments.